gammapy/estimators/metadata.py

```diff
@@ -0,0 +1,256 @@
+        elif isinstance(v, CreatorMetaData):
+            return v
+        else:
+            raise ValidationError(
+                f"Incorrect [creation]. Expect 'CreatorMetaData' got {type(v)} instead."
+            )
+
+    @classmethod
+    def from_default(cls):
+        return cls(
+            # creation=CreatorMetaData.from_default(), target_position=None, gti=None
+            creation=CreatorMetaData.from_default()
+        )
+
+    @classmethod
+    def from_dict(cls, data):
+        """Extract metadata from a dictionary.
+
+        Parameters
+        ----------
+        data : dict
+            Dictionary containing data.
+        """
+
+        sed_type = data["SED_TYPE"] if "SED_TYPE" in data else None
+        sed_type_init = data["SEDTYPEI"] if "SEDTYPEI" in data else None
+        creation = CreatorMetaData.from_dict(data)
+        meta = cls(sed_type=sed_type, sed_type_init=sed_type_init, creation=creation)
+
+        if "UL_CONF" in data:
+            meta.ul_conf = float(data["UL_CONF"])
+        if "N_SIGMA" in data:
+            meta.n_sigma = float(data["N_SIGMA"])
+
+        if "NSIGMAUL" in data:
+            meta.n_sigma_ul = float(data["NSIGMAUL"])
+            if meta.ul_conf and meta.n_sigma_ul != np.round(
+                stats.norm.isf(0.5 * (1 - meta.ul_conf)), 1
+            ):
+                log.warning(
+                    f"Inconsistency between n_sigma_ul={meta.n_sigma_ul} and ul_conf={meta.ul_conf}"
+                )
+        elif meta.ul_conf:
+            meta.n_sigma_ul = np.round(stats.norm.isf(0.5 * (1 - meta.ul_conf)), 1)
+
+        if "STSTHUL" in data:
+            meta.sqrt_ts_threshold_ul = float(data["STSTHUL"])
+        if "NSIGMSEN" in data:
+            meta.n_sigma_sensitivity = float(data["NSIGMSEN"])
+        if "TARGETNA" in data:
+            meta.target_name = data["TARGETNA"]
+        if "OBS_IDS" in data:
+            meta.obs_ids = FluxMetaData._obsids_from_string(data["OBS_IDS"])
+        if "DATASETS" in data:
+            meta.dataset_names = data["DATASETS"]
+        if "INSTRU" in data:
+            meta.instrument = data["INSTRU"]
+        if "TARGETPO" in data:
+            meta.target_position = FluxMetaData._target_from_string(data["TARGETPO"])
+
+        for kk in data:
+            if kk not in STANDARD_KEYS:
+                if meta.optional is None:
+                    meta.optional = {}
+                meta.optional[str(kk).lower()] = data[kk]
+
+        return meta
+
+    def to_table(self, table):
+        """Write the metadata into a table. Only the non-null information are stored.
+
+        Parameters
+        ----------
+        table : `~astropy.table.Table`
+            Flux table.
+        """
+
+        if self.sed_type:
+            table.meta["SED_TYPE"] = self.sed_type
+        if self.sed_type_init:
+            table.meta["SEDTYPEI"] = self.sed_type_init
+        if self.ul_conf:
+            table.meta["UL_CONF"] = str(self.ul_conf)
```

Comment:
The type conversion her is not needed, pedantic will guarantee that it's as thing already...

Reply:
It is not in the wrong direction? Here, we fill an astropy table, not a pedantic object. I think that the casting is needed